internal/app/triggerfactory.go

```diff
@@ -84,7 +105,7 @@ func (svc *Service) processMessageOnRuntime(envelope types.MessageEnvelope) erro
 	context := svc.buildContextForRuntime(envelope)
 	defaultPipeline := svc.runtime.GetDefaultPipeline()
 
-	if defaultPipeline != nil { // then we aren't configured for topic matching, use default
+	if len(defaultPipeline.Transforms) > 0 { // then we aren't configured for topic matching, use default
```

Comment:
I like they way you previously had it where it didn't matter if they used Set or Add or both. Just processed any matching pipelines.